bugRzilla_review.Rmd

```diff
@@ -168,45 +215,659 @@ Also there are columns which are empty so they are of no use of the analysis lik
-fig <- plot_ly(data, y = ~bug_id, x = ~creation, type = 'scatter', mode = 'markers')
-fig
+```{r Bug changing was last emailed to the cc list graph}
+# Plotting the Time Series graph with the bug_id and delta_ts
+last_modified_graph <- bugs_df %>%
+    ggplot(aes(x = delta_ts, y = bug_id)) +
+    geom_point() +
+    labs(title = "Bug changing was last emailed to the cc list",
+           y = "Bug ID",
+           x = "Date") + theme_bw(base_size = 15)
+ggplotly(last_modified_graph)
+```
 
-data <- data.frame(bug_id, delta)
-fig <- plot_ly(data, y = ~bug_id, x = ~delta, type = 'scatter', mode = 'markers')
-fig
+<font size="4">
+    The `Time-series graph` shows that which bug_id was the last update. Most of the bugs are last updated in the month of `January`,`March`, `April`, and `July`.
+</font>
 
-data <- data.frame(bug_id, deadline)
-fig <- plot_ly(data, y = ~bug_id, x = ~deadline, type = 'scatter', mode = 'markers')
-fig
+```{r Bug Resolution Bar graph}
+Resolution_graph <- ggplot(bugs_df,aes(x = resolution)) +
+    geom_bar() +
+    scale_x_discrete(guide = guide_axis(n.dodge = 5)) +
+    labs(
+        title = "Bug Resolution Bar graph with Bug Count",
+        x = "Resolution",
+        y = "Bug Count"
+    ) + coord_flip()
+ggplotly(Resolution_graph)
+```
 
-# bar plot with bug_id and resolution 
-data <- data.frame(bug_id, resolution)
-fig <- plot_ly(data, x = ~resolution, y = ~bug_id, name = 'creation', type = 'bar')
-fig
+<font size="4">
+    The `Resolution bar-graph` shows which bug_id belongs to which resolution category, if the bug is in a closed state, e.g. FIXED, DUPLICATE, etc. As we can conclude, that most bugs belong to the fixed category of the resolution.
+</font>
 
+```{r Bug Status Bar graph}
+Status_graph <- ggplot(bugs_df,aes(x = bug_status)) +
+    geom_bar() +
+    scale_x_discrete(guide = guide_axis(n.dodge = 5)) +
+    labs(
+        title = "Bug Status Bar graph with Bug Count",
+        x = "Bug Status",
+        y = "Bug Count"
+    )
+ggplotly(Status_graph)
+```
+<font size="4">
+    The `bug_status bar-graph` shows which bug_id belongs to which category of bug_status, e.g. NEW, RESOLVED, etc. As we can conclude, that most bugs belong to the closed category of the bug_status.
+</font>
+
+```{r Bug Severity Bar graph}
+Severity_graph <- ggplot(bugs_df,aes(x = bug_severity)) +
+    geom_bar() +
+    scale_x_discrete(guide = guide_axis(n.dodge = 5)) +
+    labs(
+        title = "Bug Severity Bar graph with Bug Count",
+        x = "Bug Severity",
+        y = "Bug Count"
+    )
+ggplotly(Severity_graph)
+```
+
+<font size="4">
+    The `bug_severity bar-graph` shows which bug_id belongs to which category of bug_severity. Most of the bugs which are filed are normal, some of the bugs that are filled under enhancements are retested for some minor and major features, and a very few bugs are filed under the blocker category.
+</font>
+
+## Data Exploartion of `bugs` and `Attachments` Table from the Database
```

Comment:
Is this data analysed anywhere? How many bug reports have a patch attached? How often is a patch attached by the original submitter and how often by someone else? What do people attach that's not a patch?

Reply:
Yes, All the `bar graphs` are analyzed using the `bugs` and `Attachments` Table from the Database.

Reply:
It is the attachments table I was referring to here. I didn't see any plots/exploration of the bug report attachments beyond the skimr summary of the table. It would be interesting to find out the the answers to the questions I raised in my comment above (How many bug reports have a patch attached?, etc).

Reply:
This can be an interesting plot to see. I'll look into it.